h/static/scripts/group-forms/config.ts

```diff
@@ -8,6 +8,18 @@ export type ConfigObject = {
   styles: string[];
   api: {
     createGroup: APIConfig;
+    updateGroup: APIConfig;
+  };
+  components: {
+    CreateEditGroupForm: {
+      mode: string;
+      group: {
+        pubid: string;
+        name: string;
+        description: string;
+        link: string;
+      };
```

Comment:
All the group fields are empty for a new group, so an alternative way to represent this would be to simply make the `group` field null when creating a new group and omit the `mode` in that case. Something like:

```diff
diff --git a/h/static/scripts/group-forms/components/CreateGroupForm.tsx b/h/static/scripts/group-forms/components/CreateGroupForm.tsx
index d7013a3c2..6bd3fbcdb 100644
--- a/h/static/scripts/group-forms/components/CreateGroupForm.tsx
+++ b/h/static/scripts/group-forms/components/CreateGroupForm.tsx
@@ -120,11 +120,10 @@ function TextField({
 
 export default function CreateGroupForm() {
   const config = useMemo(() => readConfig(), []);
-  const mode = config.components.CreateEditGroupForm.mode;
-  const group = config.components.CreateEditGroupForm.group;
+  const group = config.group;
 
-  const [name, setName] = useState(group.name);
-  const [description, setDescription] = useState(group.description);
+  const [name, setName] = useState(group?.name ?? '');
+  const [description, setDescription] = useState(group?.description ?? '');
   const [errorMessage, setErrorMessage] = useState('');
   const [saving, setSaving] = useState(false);
 
@@ -192,16 +191,16 @@ export default function CreateGroupForm() {
   };
 
   let onSubmit;
-  if (mode === 'create') {
-    onSubmit = createGroup;
-  } else {
+  if (group) {
     onSubmit = updateGroup;
+  } else {
+    onSubmit = createGroup;
   }
 
   return (
     <div className="text-grey-6 text-sm/relaxed">
       <h1 className="mt-14 mb-8 text-grey-7 text-xl/none">
-        {mode === 'create'
+        {!group
           ? 'Create a new private group'
           : 'Edit group details'}
       </h1>
@@ -245,14 +244,14 @@ export default function CreateGroupForm() {
             disabled={saving}
             data-testid="button"
           >
-            {mode === 'create' ? 'Create group' : 'Save changes'}
+            {!group ? 'Create group' : 'Save changes'}
           </Button>
         </div>
       </form>
 
       <footer className="mt-14 pt-4 border-t border-t-text-grey-6">
         <div className="flex">
-          {mode === 'edit' && (
+          {group && (
             <a href={group.link}>← Back to group overview page</a>
           )}
           <div className="grow" />
diff --git a/h/static/scripts/group-forms/config.ts b/h/static/scripts/group-forms/config.ts
index 34c43db0c..22230971c 100644
--- a/h/static/scripts/group-forms/config.ts
+++ b/h/static/scripts/group-forms/config.ts
@@ -10,17 +10,12 @@ export type ConfigObject = {
     createGroup: APIConfig;
     updateGroup: APIConfig;
   };
-  components: {
-    CreateEditGroupForm: {
-      mode: string;
-      group: {
-        pubid: string;
-        name: string;
-        description: string;
-        link: string;
-      };
-    };
-  };
+  group: {
+    pubid: string;
+    name: string;
+    description: string;
+    link: string;
+  } | null;
 };
 
 /** Return the frontend config from the page's <script class="js-config">. */
diff --git a/h/templates/groups/create_edit.html.jinja2 b/h/templates/groups/create_edit.html.jinja2
index 713868e90..cae4d5f55 100644
--- a/h/templates/groups/create_edit.html.jinja2
+++ b/h/templates/groups/create_edit.html.jinja2
@@ -15,17 +15,7 @@
           "url": "{{ request.route_url("api.group", id=group.pubid) }}"
         }
       },
-      "components": {
-        "CreateEditGroupForm": {
-          "mode": "{{ mode }}",
-          "group": {
-            "pubid": "{{ group.pubid }}",
-            "name": "{{ group.name }}",
-            "description": "{{ group.description }}",
-            "link": "{{ group.link }}"
-          }
-        }
-      }
+      "group": {{ group|tojson }}
     }
   </script>
 {% endblock %}
diff --git a/h/views/groups.py b/h/views/groups.py
index 73a449f94..8a810e8c4 100644
--- a/h/views/groups.py
+++ b/h/views/groups.py
@@ -20,13 +20,7 @@ class GroupCreateEditController:
         """Render the page for creating a new group."""
         return {
             "page_title": "Create a new private group",
-            "mode": "create",
-            "group": {
-                "pubid": "",
-                "name": "",
-                "description": "",
-                "link": "",
-            },
+            "group": None
         }
 
     @view_config(
@@ -38,7 +32,6 @@ class GroupCreateEditController:
 
         return {
             "page_title": "Edit group details",
-            "mode": "edit",
             "group": {
                 "pubid": group.pubid,
                 "name": group.name,
```

The upside is that this explicitly models the possibility that there is no group, so e.g. it's harder to accidentally write code which uses `pubid` in a context where it may be empty.

The downside of this approach is that if you wanted to provide default values for fields, those would need to be defined in the frontend, or provided via a separate field.

Reply:
:+1: Done, I've removed the `mode` field in favor of reacting to `group` being truthy or falsey